src/google-analytics/GoogleAnalyticsService.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * @implements {GoogleAnalyticsService}
+ * @memberof module:GoogleAnalytics
+ */
+class GoogleAnalyticsService {
+  constructor({ config }) {
+    this.analyticsId = config.GOOGLE_ANALYTICS_4_ID;
+
+    if (this.analyticsId) {
+      this.initializeService();
+    }
+  }
+
+  initializeService() {
+    // Create a queue, but don't obliterate an existing one!
+    global.googleAnalytics = global.googleAnalytics || [];
+    const { googleAnalytics } = global;
+
+    // If the snippet was invoked do nothing.
+    if (googleAnalytics.invoked) {
+      return;
+    }
+
+    // Invoked flag, to make sure the snippet
+    // is never invoked twice.
+    googleAnalytics.invoked = true;
+
+    // Define a method to load Analytics.js,
+    // and that will be sure to only ever load it once.
+    googleAnalytics.load = (key, options) => {
+      // Create an async script element based on your key.
+      const scriptSrc = document.createElement('script');
+      scriptSrc.type = 'text/javascript';
+      scriptSrc.onerror = () => {
+        const event = new Event('segmentFailed');
+        document.dispatchEvent(event);
+      };
+      scriptSrc.async = true;
+      scriptSrc.src = `https://www.googletagmanager.com/gtag/js?id=${key}`;
+
+      const script = document.createElement('script');
```

Comment:
maybe it is worth naming the way like above to specify the exact destination of the script, what do you think?

Reply:
@UvgenGen Please make all variable names meaningful, to describe the purposes of their values